src/pages/index.js

```diff
@@ -36,36 +45,82 @@ const editProfilePopup = new PopupWithForm(
   "#edit-profile-popup",
   handleProfileSave
 );
+const editProfileImagePopup = new PopupWithForm(
+  "#edit-profile-image-popup",
+  handleProfileImageSave
+);
 const addCardPopup = new PopupWithForm("#add-card-popup", handleAddCard);
+const confirmDeletePopup = new PopupConfirm('#confirm-delete-popup', handleDeleteCard);
 
 /*═══════╕
  │ CARDS │
  ╘═══════*/
-const cardSection = new Section(
-  { items: initialCards, renderer: handleCreateCard },
-  ".cards__list"
-);
+let cardSection;
+let targetCard;
 
 /*════════════════╕
  │ EVENT HANDLERS │
  ╘════════════════*/
 const imagePopup = new PopupWithImage("#image-popup");
+const deleteButton = document.querySelector("#confirm-delete-button");
 
 function handleAddCard({ title, url }) {
-  const card = {
-    name: title,
-    link: url,
-  };
-  cardSection.addItem(handleCreateCard(card));
+  api.addCard({name: title, link: url})
+  .then(cardObj => {
+    cardSection.addItem(cardObj);
+    addCardPopup.close();
+  })
+  .catch(err => console.error(err));
 }
 function handleCreateCard(card) {
-  return new Card(card, "#card-template", handleImageClick).getElement();
+  return new Card(card, "#card-template", handleImageClick, handleDeleteClick, handleLikeClick).getElement();
+}
+function handleDeleteCard() {
+  confirmDeletePopup.deleting(true);
+  api.deleteCard(targetCard)
+    .then(() => api.getPageInfo()
+      .then(([cardArray]) => {
+        console.log(cardArray);
+        cardSection.setItems(cardArray);
+        cardSection.renderItems();
+        confirmDeletePopup.close();
+      })
+    )
+    .catch(err => console.log(err));
+}
+function handleDeleteClick(card) {
+  targetCard = card.getData();
+  confirmDeletePopup.deleting(false);
+  confirmDeletePopup.open();
 }
 function handleImageClick(card) {
   imagePopup.open(card.getData());
 }
+function handleLikeClick(card) {
+  targetCard = card.getData();
+  api.toggleLike(targetCard)
+  .then(cardSection.renderItems())
+  .catch(err => console.log(err));
+}
+function handleProfileImageSave({url}) {
+  console.log(url);
+  editProfileImagePopup.saving(true);
+  profileImage.src = url;
```

Comment:
`profileImage.src = url;`

You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error it wouldn’t change